curriculum/challenges/english/25-front-end-development/lab-contact-form/66f26c32ec6f90df01a44f60.md

```diff
@@ -0,0 +1,458 @@
+   - A `button` element with the text `"Submit"` and a `type` attribute of `submit`.
+
+4. Each `input` and `textarea` elements should have their corresponding `label` element with a `for` attribute set to the element's `id`.
+
+5. The `.form-container` `div` should have a background color, and have values for `border-radius`, `padding`, and `width`.
+
+6. The `label` elements should have a margin and a font color.
+
+7. The `input` and `textarea` elements should have values for:
+   
+   - `width`.
+   - `padding`.
+   - Margin on the bottom.
+
+8. The `button` element should have a:
+   
+   - Background color
+   - Font size apart from the default.
+
+9.  The button should have a hover effect that changes the background color.
+
+**Note:** Be sure to link your stylesheet in your HTML and apply your CSS.
+
+# --hints--
+
+You should have a `div` with a class of `form-container`.
+
+```js
+const el = document.querySelector('div.form-container');
+assert.isNotNull(el);
+```
+
+You should have a `form` element within the `.form-container` `div`.
+
+```js
+const el = document.querySelector('.form-container form');
+assert.isNotNull(el);
+```
+
+Within the `form`, you should have an `h2` element with some text.
+
+```js
+const el = document.querySelector('.form-container form h2');
+assert(!!el && el.innerText.length > 0);
+```
+
+Within the `form`, you should have a text input field with the type set to `text`.
+
+```js
+const el = document.querySelector('.form-container form input[type="text"]');
+assert.isNotNull(el);
+```
+
+Your `type=text` input field should have an `id`, a `name`, and a `required` attribute.
+
+```js
+const el = document.querySelector('.form-container form input[type="text"]');
+
+assert(!!el && el.required);
+assert.isNotEmpty(el?.getAttribute('id'));
+assert.isNotEmpty(el?.getAttribute('name'));
+```
+
+Within the `form`, you should have an email input field with the `type` set to `email`.
+
+```js
+const el = document.querySelector('.form-container form input[type="email"]');
+assert.isNotNull(el);
+```
+
+Your `type=email` input field should have an `id`, a `name`, and a `required` attribute.
+
+```js
+const el = document.querySelector('.form-container form input[type="email"]');
+assert(!!el && el.required);
+assert.isNotEmpty(el?.getAttribute('id'));
+assert.isNotEmpty(el?.getAttribute('name'));
+```
+
+Within the `form`, you should have a `textarea` with an `id`, `name`, and a `required` attribute.
+
+```js
+const el = document.querySelector('.form-container form textarea');
+assert.isNotNull(el);
+assert.isNotNull(document.querySelector('.form-container form textarea[required]'));
+assert.isNotEmpty(el?.getAttribute('id'));
+assert.isNotEmpty(el?.getAttribute('name'));
+```
+
+Within the `form`, you should have a `button` element with a `type` attribute of `submit`.
+
+```js
+const el = document.querySelector('.form-container form button');
+assert(el);
```

Comment:
```suggestion
assert.exists(el);
```

Reply:
or also just remove this line, this test is implicit in the `el?` in the next line

Reply:
Removed this.